### DISCUSSION TITLE
Bell Fibe TV

### DISCUSSION THREAD

**Discussion by jdrom** [fabianishere/udm-iptv#9] [Q&A]
I'm attempting to use udm-iptv with the new UDMP 1.11.0-14 beta now that there's built in kernel support for it. It looks promising but running into an issue where once I start up the container, it sets the iptv network as the default route and I lose connectivity to the Internet. 

On my pfSense setup, I decreased the priority of the iptv gateway so it never becomes default, on my EdgeRouter, I set the flag that said not to create a default route. It doesn't seem like udhcp offers any way to prevent that without editing the underlying script it uses? And that would lead me towards building my own image?

I'm attempting to setup Bell Canada Fibe TV, which looks something like this:
Internet: PPPoE on VLAN35 (eth9.35/ppp0 - Bell GPON SFP plugged into an ES-16-XG and DAC to UDMP)
IPTV: DHCP on VLAN36 - internal to Bell network

And using these settings for the container:
IPTV_WAN_INTERFACE="eth9" 
IPTV_WAN_RANGES="10.2.0.0/16 10.125.0.0/16" 
IPTV_WAN_VLAN="36" 
IPTV_WAN_DHCP_OPTIONS="" 
IPTV_LAN_INTERFACES="br40" 
IPTV_LAN_RANGES="192.168.2.0/24"

As an aside, it looks like IPTV_LAN_RANGES is only added as an upstream altnet in igmpproxy? I can't see any other place it's referenced. I've never put my LAN subnet as an altnet before, so I'm leaning towards removing this. 

Thanks for your great work so far! I didn't think this warranted an issue as I'm not sure if you plan on generalizing this for the world, or if it will remain focused on KPN. It's a solid starting point either way.

**Comment by fabianishere** [comment]
I suspect you need to add `-O staticroutes` to your DHCP options, which will prevent the default route from being added.

**Reply by fabianishere** [reply]
Once you get this setup working, I might document your settings so other Bell Canada users can re-use your settings.

**Reply by jdrom** [reply]
For sure, I've updated the title to be focused on Bell Fibe TV as a whole. 

Using `-O staticroutes` is what seems to make a default route but my UDMP doesn't appear to have one normally. I guess UBNT must manage the routing elsewhere. I had discovered using `-o`, which is no DHCP options, seemed to work. 

That being said, while TV itself seems to work, if I attempt to use the On Demand (VOD) service, it fails to load. If I reboot the STB/PVR, they don't boot up successfully either. 

In comparing the configs between pfSense/EdgeRouter and this though, it looks like my DHCP lease is a totally different network. On the UDMP I'm getting a 10.125.176.0/21 address, whereas on pfSense/ER, I get 10.238.0.0/20. I've never had to do anything unique with DHCP options before, but I'll read up on udhcp some more as it would seem I'm getting an IP from the wrong DHCP pool somehow...

**Reply by fabianishere** [reply]
What settings are you using in pfSense/EdgeRouter currently? It might be that the DHCP server does not return a Classless Static Routes option (121), causing the default route to be added.

**Reply by jdrom** [reply]
In pfSense, I just enable the interface and tell it to use DHCP.  No advanced options set. I create a static route for 10.2.0.0/16 to the IPTV network gateway/interface. I then have 2 firewall rules on the IPTV interface, one to allow all IGMP traffic and another which allows UDP traffic from 10.2.0.0/16. 

I think I had saw some people suggest 10.125.0.0/16 was used too, so I had added it to my igmpproxy config but clearly didn't set it up entirely... was just reviewing my pfSense logs and noticed traffic was being blocked... so I've added a static route and FW rule to allow the UDP traffic from it as well now. 

To add the routes, I would append:
```
ip route add 10.2.0.0/16 dev iptv
ip route add 10.125.0.0/16 dev iptv
```
to the bottom of the 15-iptv.sh script? 

Would any firewall be at play here? I'm guessing it must be wide open though since the TV does partially work.

Maybe it's just a coincidence that I get a different IP here though, but in the past 3yr I've always had 10.238.x.x regardless of device...

**Comment by jdrom** [comment]
Not the most elegant, but this seems to be working for me at least. I'm making the assumption the gateway on Bell's end would be .1 which seems to be the case in my experience. Not sure if there's any way to get the actual gateway from the DHCP lease without changing the udhcp script within the image. 

```
tee  /mnt/data/on_boot.d/15-iptv.sh <<EOF >/dev/null
if podman container exists iptv; then
  podman rm -f iptv
fi
podman run --network=host --privileged \
    --name iptv -i -d --restart always \
    -e IPTV_WAN_INTERFACE="eth9" \
    -e IPTV_WAN_RANGES="10.2.0.0/16 10.125.0.0/16" \
    -e IPTV_WAN_VLAN="36" \
    -e IPTV_WAN_DHCP_OPTIONS="-n -o" \
    -e IPTV_LAN_INTERFACES="br40" \
    -e IPTV_LAN_RANGES=" " \
    fabianishere/udm-iptv
sleep 5
GW=$(route -n | awk '/iptv/ { print $1 }' | sed 's/.$/1/')
ip route add 10.2.0.0/16 via $GW
ip route add 10.125.0.0/16 via $GW
EOF
chmod +x /mnt/data/on_boot.d/15-iptv.sh
```

**Comment by diegobrandao** [comment]
@jdrom Is this config still working for you? I am trying to setup Bell Fibe, this config that you posted, almost worked for me.
However, it only works for some seconds, then it goes down... 

I am using the UDM Base (1.11.0), and I am located in Quebec... 

My 15-iptv.sh:
```
IPTV_WAN_INTERFACE="eth4"
IPTV_WAN_RANGES="10.2.0.0/16 10.125.0.0/16"
IPTV_WAN_VLAN="36"
IPTV_WAN_DHCP_OPTIONS="-n -o"
IPTV_LAN_INTERFACES="br36"
IPTV_LAN_RANGES=""
IPTV_IGMPPROXY_DEBUG="true"

if podman container exists iptv; then
  podman rm -f iptv
fi
podman run --network=host --privileged \
    --name iptv -i -d --restart always \
    -e IPTV_WAN_INTERFACE="$IPTV_WAN_INTERFACE" \
    -e IPTV_WAN_RANGES="$IPTV_WAN_RANGES" \
    -e IPTV_WAN_VLAN="$IPTV_WAN_VLAN" \
    -e IPTV_WAN_DHCP_OPTIONS="$IPTV_WAN_DHCP_OPTIONS" \
    -e IPTV_LAN_INTERFACES="$IPTV_LAN_INTERFACES" \
    -e IPTV_LAN_RANGES="$IPTV_LAN_RANGES" \
    -e IPTV_IGMPPROXY_DEBUG="$IPTV_IGMPPROXY_DEBUG=" \
    fabianishere/udm-iptv -d -v
sleep 5
GW=$(route -n | awk '/iptv/ { print $1 }' | sed 's/.$/1/')
ip route add 10.2.0.0/16 via $GW
ip route add 10.125.0.0/16 via $GW
```


Logs:
```
# podman logs -f iptv
udm-iptv: Obtaining IP address for VLAN interface...
udhcpc: started, v1.33.1
udhcpc: sending discover
udhcpc: sending select for 10.152.*.*
udhcpc: lease of 10.152.*.* obtained, lease time 2329
udm-iptv: NATing IPTV network ranges (if necessary)...
udm-iptv: Setting up igmpproxy..
udm-iptv: Starting igmpproxy..
adding VIF, Ix 0 Fl 0x0 IP 0x0103a8c0 br36, Threshold: 1, Ratelimit: 0
adding VIF, Ix 1 Fl 0x0 IP 0xd359980a iptv, Threshold: 1, Ratelimit: 0
Joining group 224.0.0.2 on interface br36
Joining group 224.0.0.22 on interface br36
RECV Membership query   from 192.168.3.1     to 224.0.0.1
RECV V2 member report   from 192.168.3.1     to 224.0.0.251
The IGMP message was from myself. Ignoring.
RECV Membership query   from 192.168.3.1     to 224.0.0.1
RECV V2 member report   from 10.152.89.211   to 224.0.0.251
The IGMP message was from myself. Ignoring.
RECV V2 member report   from 192.168.3.1     to 224.0.0.22
```

@fabianishere Thanks for you work!! 

Do you have ideas of what could be the issue?
Thanks,
Diego

**Reply by jdrom** [reply]
I'd suggest changing the local network on your UDM to be something other than VLAN36, wondering if there's some weird overlaps having the same VLAN on both the LAN and "WAN" side. Also is IGMP Snooping enabled on the network in UniFi? If you have your Fibe TV boxes plugged into a non-UniFi switch downstream, it would also need IGMP Snooping enabled. 

And since your DHCP lease is 10.152.x.x, you should probably change the 10.125.0.0/16 route to match. I don't think it's even necessary, but some people have said to keep a route to Bell's DHCP server.

**Comment by dcrompto** [comment]
hi @jdrom @fabianishere,

Based out of Ottawa, Ontario.

I am up and running with the STB without issue for normal TV watching and on demand working, but when I try and play any PVR content it comes up with a blank screen.

Have the LAN in the unifi UI marked as VLAN 40, with IGMP Snooping turned on

my udm-iptv.sh

```
# Interface on which IPTV traffic enters the router
IPTV_WAN_INTERFACE="eth9"
# ID of VLAN which carries IPTV traffic (use 0 if no VLAN is used)
IPTV_WAN_VLAN="36"
# Name of the IPTV VLAN interface
IPTV_WAN_VLAN_INTERFACE="iptv"
# IP ranges from which the IPTV traffic originates (separated by spaces)
IPTV_WAN_RANGES="10.2.0.0/16 10.125.0.0/16"
# DHCP options to send when requesting an IP address
IPTV_WAN_DHCP_OPTIONS="-n -o"
# LAN interfaces on which IPTV should be made available
IPTV_LAN_INTERFACES="br40"
# Disable quickleave for igmpproxy
IPTV_IGMPPROXY_DISABLE_QUICKLEAVE="false"
# Enable debugging for igmpproxy
IPTV_IGMPPROXY_DEBUG="false"
```

my udm-iptv-diag

```
Please share the following output with the developers:
 === Configuration ===
WAN Interface: eth9
WAN Ranges: 10.2.0.0/16 10.125.0.0/16
WAN VLAN: 36 (dev iptv)
WAN DHCP options: -n -o
LAN Interfaces: br40
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
33: iptv@eth9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue qlen 1000
     inet 10.121.64.X/19 scope global iptv
       valid_lft forever preferred_lft forever
10.2.0.0/16 via 10.121.64.1 dev iptv 
10.121.0.0/16 via 10.121.64.1 dev iptv 
10.121.64.0/19 dev iptv  src 10.121.64.X 
=== Service Logs ===
-- Logs begin at Mon 2022-03-21 12:21:12 EDT, end at Mon 2022-03-21 21:09:05 EDT. --
-- No entries --
```

doing a tcpdump when accessing the content, seems to be trying to hit 233.89.188.1:

```
21:26:32.439440 IP 10.121.64.X.43487 > 233.89.188.1.10001: UDP, length 4
21:26:32.439509 IP 10.121.64.X.43487 > 255.255.255.255.10001: UDP, length 4
21:26:35.417385 IP 10.121.64.X.46066 > 233.89.188.1.10001: UDP, length 4
21:26:35.417657 IP 10.121.64.X.46066 > 255.255.255.255.10001: UDP, length 4
21:26:35.418105 IP 10.121.64.X.10001 > 233.89.188.1.10001: UDP, length 4
21:26:35.418315 IP 10.121.64.X.10001 > 255.255.255.255.10001: UDP, length 4
21:26:36.312239 IP 10.121.64.X.43505 > 10.2.90.33.43962: UDP, length 113
21:26:36.321023 IP 10.2.90.33.43962 > 10.121.64.X.43505: UDP, length 50
21:26:42.440701 IP 10.121.64.X.43487 > 233.89.188.1.10001: UDP, length 4
21:26:42.440769 IP 10.121.64.X.43487 > 255.255.255.255.10001: UDP, length 4
21:26:44.106613 IP 0.0.0.0 > all-systems.mcast.net: igmp query v2
21:26:45.426369 IP 10.121.64.X > 224.0.0.251: igmp v2 report 224.0.0.251
21:26:45.979826 IP 10.121.64.X.46066 > 233.89.188.1.10001: UDP, length 4
21:26:45.979896 IP 10.121.64.X.46066 > 255.255.255.255.10001: UDP, length 4
21:26:45.980150 IP 10.121.64.X.10001 > 233.89.188.1.10001: UDP, length 4
21:26:45.980243 IP 10.121.64.X.10001 > 255.255.255.255.10001: UDP, length 4
```

Recently new to this so any other tracing I can do happy to try out it.

thanks for the help

**Reply by jdrom** [reply]
I'd probably change the 10.125.0.0/16 range to 10.121.0.0/16 to match the subnet you're getting on the DHCP lease.

Does the PVR boot up successfully?

**Reply by dcrompto** [reply]
thanks for the suggestion, will give that a try shortly

The PVR / STB boots up without issue, Live TV and On Demand content works fine, only thing not working is trying to play recorded PVR content on the STB. it just opens up the content and doesn't play. Not the worst to live without but curious about to get it working.

The TCP dump above is when it is trying to play the content.

**Reply by dcrompto** [reply]
That worked! *facepalm* seems so simple

final setup, for anyone else who might need it

udm-iptv.conf

```
# Interface on which IPTV traffic enters the router
IPTV_WAN_INTERFACE="eth9"
# ID of VLAN which carries IPTV traffic (use 0 if no VLAN is used)
IPTV_WAN_VLAN="36"
# Name of the IPTV VLAN interface
IPTV_WAN_VLAN_INTERFACE="iptv"
# IP ranges from which the IPTV traffic originates (separated by spaces)
IPTV_WAN_RANGES="10.2.0.0/16 10.121.0.0/16"
# DHCP options to send when requesting an IP address
IPTV_WAN_DHCP_OPTIONS="-n -o"
# LAN interfaces on which IPTV should be made available
IPTV_LAN_INTERFACES="br40"
# Disable quickleave for igmpproxy
IPTV_IGMPPROXY_DISABLE_QUICKLEAVE="false"
# Enable debugging for igmpproxy
IPTV_IGMPPROXY_DEBUG="false"
```

routes added

```
GW=$(route -n | awk '/iptv/ { print $1 }' | sed 's/.$/1/')
ip route add 10.2.0.0/16 via $GW
ip route add 10.121.0.0/16 via $GW
```

**Reply by jdrom** [reply]
I guess the tip would be to make sure the 2nd range/route matches what you get in your DHCP lease. The 10.2.0.0/16 seems pretty universal though.

If we base it on these posts, possible examples would be:
Toronto - 10.125.0.0/16 (though I previously used to get 10.238.0.0/16)
Ottawa - 10.121.0.0/16
Quebec - 10.152.0.0/16

And I don't believe the "-n" is needed in DHCP options, I know I had posted that originally, but I've just been using "-o" now.

**Reply by diegobrandao** [reply]
It is working for me as well. I am based on Quebec, Quebec.

However, I was not able to make it work in another LAN, I tried with (br40), but I got the same issue I had before. It only worked for some seconds... It is probably something in the config of this lan... In any case, for now, I don't really need to put it in another LAN. 

udm-iptv.conf
```
## Example configuration for udmp-iptv ##
# Interface on which IPTV traffic enters the router
IPTV_WAN_INTERFACE="eth4"
# ID of VLAN which carries IPTV traffic (use 0 if no VLAN is used)
IPTV_WAN_VLAN="36"
# Name of the IPTV VLAN interface
IPTV_WAN_VLAN_INTERFACE="iptv"
# IP ranges from which the IPTV traffic originates (separated by spaces)
IPTV_WAN_RANGES="10.2.0.0/16 10.152.0.0/16"
# DHCP options to send when requesting an IP address
IPTV_WAN_DHCP_OPTIONS="-o"
# LAN interfaces on which IPTV should be made available
IPTV_LAN_INTERFACES="br0"
# Disable quickleave for igmpproxy
IPTV_IGMPPROXY_DISABLE_QUICKLEAVE="false"
# Enable debugging for igmpproxy
IPTV_IGMPPROXY_DEBUG="false"
```

routes added
```
GW=$(route -n | awk '/iptv/ { print $1 }' | sed 's/.$/1/')
ip route add 10.2.0.0/16 via $GW
ip route add 10.152.0.0/16 via $GW
```

Thanks @jdrom and @dcrompto for posting your config it helped a lot.

**Comment by Jason112112** [comment]
JDROM - would you mind providing the firewall rules you added?  I think I have everything else in your post working properly, but its still not working for me.

And thank you for the very detailed information you've posted!! :)

Thanks!!

**Reply by jdrom** [reply]
Shouldn't be any firewall rules needed, what behaviour are you seeing? Outside of fabian's work here, it should only be adding the 2 routes (10.2.0.0/16 and one that matches the first 2 octets of your DHCP lease) and then to make sure you are using Bell's DNS. 

I'm not using this anymore though, since I upgraded to 3Gbps, so just using the HH4K directly for TV now.

**Comment by Jason112112** [comment]
Thanks for the info.  I guess I must not be getting the proper gateway.

I should note that I'm in atlantic Canada which does things a little different.

I'm getting an IPTV IP, and I've added the routes, but the STB still won't even boot - it gets stuck at the blue Fibe screen.

```
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth9
WAN VLAN: 34 (dev iptv)
WAN DHCP: true (options "-o")
WAN Ranges: 10.2.0.0/16 10.224.0.0/16
LAN Interfaces: br40
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
31: iptv@eth9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.198.228.70/19 scope global iptv
       valid_lft forever preferred_lft forever
10.198.224.0/19 proto kernel scope link src 10.198.228.70
=== Service Logs ===
Jul 09 10:09:06 ubnt systemd[1]: Started IPTV support for the UniFi Dream Machine.
Jul 09 10:09:06 ubnt udm-iptv[10127]: Device iptv already exists.. Deleting device
Jul 09 10:09:06 ubnt udm-iptv[10127]: Obtaining IP address for VLAN interface
Jul 09 10:09:06 ubnt udm-iptv[10127]: udhcpc (v1.22.1) started
Jul 09 10:09:06 ubnt udm-iptv[10127]: Sending discover...
Jul 09 10:09:06 ubnt udm-iptv[10127]: Sending select for 10.198.228.70...
Jul 09 10:09:06 ubnt udm-iptv[10127]: Lease of 10.198.228.70 obtained, lease time 68400
Jul 09 10:09:06 ubnt udm-iptv[10127]: NATing IPTV network ranges (if necessary)
Jul 09 10:09:06 ubnt udm-iptv[10127]: Setting up igmpproxy
Jul 09 10:09:06 ubnt udm-iptv[10127]: Starting igmpproxy
```
I used 10.198.224.1 as the gateway based on this:

```
root@ubnt:/# route -n | awk '/iptv/ { print $1 }'
10.2.0.0
10.198.0.0
10.198.224.0
```

I also saw [another guide](https://www.idscomm.ca/blog/bell-fibe-internet-iptv-with-pfsense) specific to atlantic Canada where they got the gateway IP by capturing packets and seeing which server the IPTV interface gets its DHCP lease from, which was 10.198.224.2.  This gateway didn't work for me either.

**Reply by jdrom** [reply]
I think you might have a typo in the WAN Ranges? Try with  10.198.0.0/16 instead of 10.224.0.0/16.
